OrbitLinuxTracing/PerfEventRingBuffer.cpp

```diff
@@ -121,8 +121,11 @@ void PerfEventRingBuffer::SkipRecord(const perf_event_header& header) {
   WriteRingBufferTail(metadata_page_, new_tail);
 }
 
-void PerfEventRingBuffer::ConsumeRecord(const perf_event_header& header, void* record) {
-  ReadAtTail(static_cast<uint8_t*>(record), header.size);
+void PerfEventRingBuffer::ConsumeRecord(const perf_event_header& header, void* record,
+                                        size_t size) {
+  // Sanity check so that we do not over-read from the ring-buffer and end up with random data.
+  CHECK(header.size >= size);
+  ReadAtTail(static_cast<uint8_t*>(record), size);
   SkipRecord(header);
```

Comment:
Hi Dimitry, it seems we are reading "sizeof(record)" but then skipping "sizeof(header)", is this normal?

Reply:
the size of the record includes the header.

Reply:
oh, sorry misread you comment - yes this is normal. Since we do not always want to read the whole record because of duplicate sample_id at the end.

Reply:
Ah ok I see, thanks.

Reply:
"we do not always want to read the whole **header**" right?

Reply:
record, the header contains size information about the whole record, SkipRecord also skips the whole record based on the information from the header.

Reply:
Hmm, this is confusing. header.size >= sizeof(record). We are reading "sizeof(record)" bytes, so we are not always reading the full *header*, no? Then we are calling "SkipRecord" which uses header.size internally, maybe it should be called SkipHeader?

Reply:
header.size is size of record, the header itself is always fixed size of... some even number of bytes..

Reply:
Ah ok ok, I see.